kernel/userspace.c

```diff
@@ -192,25 +191,14 @@ extern struct z_object *z_object_gperf_find(const void *obj);
 extern void z_object_gperf_wordlist_foreach(_wordlist_cb_func_t func,
 					     void *context);
 
-static bool node_lessthan(struct rbnode *a, struct rbnode *b);
-
-/*
- * Red/black tree of allocated kernel objects, for reasonably fast lookups
- * based on object pointer values.
- */
-static struct rbtree obj_rb_tree = {
-	.lessthan_fn = node_lessthan
-};
-
 /*
  * Linked list of allocated kernel objects, for iteration over all allocated
  * objects (and potentially deleting them during iteration).
  */
 static sys_dlist_t obj_list = SYS_DLIST_STATIC_INIT(&obj_list);
 
 /*
- * TODO: Write some hash table code that will replace both obj_rb_tree
- * and obj_list.
+ * TODO: Write some hash table code that will replace obj_list.
```

Comment:
There is a hash table implementation if you are looking for one [hash_map_api.h](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/hash_map_api.h)